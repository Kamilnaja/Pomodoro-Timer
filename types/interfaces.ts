export interface Registration extends Login {
  email: string;
}

export interface Login {
  login: string;
  password: string;
}

export interface LoginResponse {
  token: string;
}

export interface AuthError {
  code: ErrorCodes;
  message: string;
}

export enum ErrorCodes {
  USER_NOT_FOUND,
  USER_CURRENTLY_EXISTS,
  PASSWORD_INCORRECT,
  OTHER_ERROR,
}
