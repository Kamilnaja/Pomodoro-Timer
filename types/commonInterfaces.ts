export interface SearchResult<T> {
  result: T;
}

export interface ResultInPeriod {
  hasPreviousPeriod: boolean;
  hasNextPeriod: boolean;
}
