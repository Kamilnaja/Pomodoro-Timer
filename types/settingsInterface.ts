import { DisplayDirection } from '../web/src/settings/store/interfaces/settingsInterfaces';

export interface Settings {
  isCookieConsentAccepted: boolean;
  isSoundEnabled: boolean;
  displayDirection: DisplayDirection;
}
