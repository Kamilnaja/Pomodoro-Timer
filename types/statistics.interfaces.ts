interface Result {
  date: string;
  count: number;
}

export default interface StatsSearchResult {
  result: Result[];
}
