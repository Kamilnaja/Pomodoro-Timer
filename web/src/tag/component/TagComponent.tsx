import { TagComponentProps } from './TagComponent.props';

export const TagComponent = (props: TagComponentProps) => <div>Hello tags</div>;
