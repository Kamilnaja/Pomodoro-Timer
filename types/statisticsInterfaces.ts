import { SearchResult } from './commonInterfaces';

/*
 * Allows for searching in current year, month or day
 *
 */
export interface SearchDate {
  year: number;
  month: number;
  day?: number;
}

export interface PomodorosDoneInDay {
  date: string;
  count: number;
}

export interface TodayStats {
  result: number;
}

export interface StatsSearchResult {
  pomodoros: PomodorosDoneInDay[];
  accountActiveFrom: Date;
}

export interface TodayStatsResult extends SearchResult<number> {}
