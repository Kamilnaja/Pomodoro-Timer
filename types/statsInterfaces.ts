import { SearchResult } from './commonInterfaces';

/*
 * Allows for searching in current year, month or day
 *
 */
export interface SearchDate {
  year: number;
  month: number;
  day?: number;
}

export interface PomodorosDoneInDay {
  date: string;
  count: number;
}

export interface TodayStats {
  result: number;
}

export interface StatsSearchResult {
  pomodoros: PomodorosDoneInDay[];
  hasPreviousPeriod: boolean;
  hasNextPeriod: boolean;
}

export interface PomodoroSearchResult extends SearchResult<Pomodoro> {}

export interface TodayStatsResult extends SearchResult<number> {}

export interface Pomodoro {
  tagText: string;
  createdAt: string;
}
