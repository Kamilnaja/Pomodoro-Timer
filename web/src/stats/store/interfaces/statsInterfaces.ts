import { Result } from '../../../../../types/statisticsInterfaces';

export type HandleGetStats = (year: number, month: number) => void;

export type StatsComponentProps = { stats: Result[]; handleGetStats: HandleGetStats };

export type StatsContainerProps = {
  handleGetStats: HandleGetStats;
  stats: StatsState;
};

export interface StatsState {
  isLoading: boolean;
  error: string;
  results: Result[];
}
