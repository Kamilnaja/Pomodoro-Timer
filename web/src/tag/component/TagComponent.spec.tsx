import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Tag } from '../../../../types/statsInterfaces';
import { createTags } from '../testing/tags.test.data';
import { TagComponent } from './TagComponent';

describe('TagComponent', () => {
  let wrapper: ShallowWrapper;
  const tags: Tag[] = createTags();
  const handleChange = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<TagComponent tags={tags} handleChange={handleChange} />);
  });

  it('should match snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
