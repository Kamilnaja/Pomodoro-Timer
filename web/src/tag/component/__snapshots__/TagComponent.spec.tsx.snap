// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagComponent should match snapshot 1`] = `
<FormGroup>
  <FormLabel
    className="text-white"
    column={false}
    htmlFor="tag"
    srOnly={false}
  >
    I'm focusing at :
  </FormLabel>
  <FormControl
    as="select"
    id="tag"
    onChange={[MockFunction]}
  >
    <option
      key="0"
      value="3"
    >
      reading
    </option>
    <option
      key="1"
      value="2"
    >
      working at own project
    </option>
    <option
      key="2"
      value="5"
    >
      exercising
    </option>
  </FormControl>
</FormGroup>
`;
